### PR TITLE
323 Ode model with waning immunity and omicron parametrization

### LINKED ISSUE #323: Implement new SECIRTS Model including waning immunity
Implement the Model introduced in https://www.medrxiv.org/content/10.1101/2024.03.01.24303602v1
which is an extension of the SECIRVVS model.

### PR DESCRIPTION
**Pull request by HenrZu**
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

-

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/DLR-SC/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [ ] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [ ] New code adheres to [coding guidelines](https://github.com/DLR-SC/memilio/wiki/Coding-guidelines)
- [ ] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [ ] Tests are added for new functionality and a local test run was successful
- [ ] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [ ] Proper attention to licenses, especially no new third-party software with conflicting license has been added

### Checks by code reviewer(s)

- [ ] Corresponding issue(s) is/are linked and addressed
- [ ] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [ ] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [ ] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc

closes #323

### CODE REVIEW

File: cpp/models/ode_secirvvs/model.h [hunk 1]
```diff
@@ -36,54 +36,65 @@ namespace osecirvvs
...
-    Flow<InfectionState::InfectedCriticalImprovedImmunity,            InfectionState::SusceptibleImprovedImmunity>>;
+    Flow<InfectionState::InfectedCriticalImprovedImmunity,            InfectionState::TemporaryImmunity2>,
+    Flow<InfectionState::SusceptibleImprovedImmunity,                 InfectionState::SusceptiblePartialImmunity>,
+    Flow<InfectionState::SusceptiblePartialImmunity,                  InfectionState::SusceptibleNaive>,
+    
+    // waning
+    Flow<InfectionState::TemporaryImmunity1,                          InfectionState::SusceptiblePartialImmunity>,
+    Flow<InfectionState::TemporaryImmunity2,                          InfectionState::SusceptibleImprovedImmunity>,
+    Flow<InfectionState::SusceptibleImprovedImmunity,                 InfectionState::SusceptiblePartialImmunity>,
+    Flow<InfectionState::SusceptiblePartialImmunity,                  InfectionState::SusceptibleNaive>>;
```
Comment (mknaranja) [inline cpp/models/ode_secirvvs/model.h:96]: Critical issue: I like the new model as an extension of the previous one but it should be ensured that we can reduce it ? I think this is not the case

File: cpp/benchmarks/flow_simulation_ode_secirvvs.h [hunk 2]
```diff
@@ -432,9 +432,9 @@ class Simulation : public Base
         for (size_t i = 0; i < num_groups; ++i) {
             double new_transmission =
                 (1 - share_new_variant) *
-                    this->get_model().parameters.template get<osecirvvs::BaseInfectiousnessB117>()[(AgeGroup)i] +
+                    this->get_model().parameters.template get<osecirvvs::BaseInfectiousness>()[(AgeGroup)i] +
```
Comment (mknaranja) [inline cpp/benchmarks/flow_simulation_ode_secirvvs.h:435]: I think we don't have this in the documentation yet. Do we?

**Comment by mknaranja** [comment]
@HenrZu Can you please comment here where you moved the content to?

**Comment by HenrZu** [comment]
Content is moved to PR #963

**Pull request closed without merging**